### PR TITLE
ci: check if api schema is outdated for all commits

### PR DESCRIPTION
**Pull request by jooola**
The commits are all pushed separately to trigger a workflow on the remote repository.

### CODE REVIEW

**Comment by jooola** [comment]
This feels really fragile. If a commit in between fails to generate the schema file, it will probably break the entire workflow run.

I wonder if we can find a better way of handle the schema file update.

I tried with a pre-commit hook and storing the schema file in the repo but this is a heavy task for pre-commit.  But it might be the best solution.

**Comment by paddatrapper** [comment]
You could add logic to skip a commit if the generation fails. But perhaps that is something that can be added later if we find that it is an issue

**Comment by jooola** [comment]
Let's have this workflow also run in the pull request and if a schema generation fails for a single commit the workflow should fails. This should make it safer once it is merged into main.

**Comment by jooola** [comment]
Ok this is ready to be merged. It should be working once it land on the main branch, hope it is at least.

Please do not squash.

The workflow will check if the schema is outdated on any commit off a PR.
And once pushed to the main branch, dispatch any commit that changed the schema file to the client repository.

File: .github/workflows/api-schema.yml [hunk 1]
```diff
@@ -3,40 +3,101 @@ name: API schema
...
     steps:
       - uses: actions/checkout@v3
-      - uses: actions/setup-python@v4
         with:
-          python-version: "3.x"
+          fetch-depth: 0
 
-      - name: Clone client repo
-        uses: actions/checkout@v3
+      - uses: actions/setup-python@v3
```
Comment (paddatrapper) [inline .github/workflows/api-schema.yml:27]: Please update this to v4 and specify the `python-version "3.x"`

**Comment by jooola** [comment]
Weird seems like a dependency update introduced a regression.

**Comment by paddatrapper** [comment]
I'm re-running the tests

**Comment by paddatrapper** [comment]
Seems to be something new? The run 2hrs ago on #1886 succeeded - https://github.com/libretime/libretime/actions/runs/2515857685

**Comment by jooola** [comment]
While working on this, the schema required to be updated from the one I generated yesterday or so. All the filtering args were removed from the schema for the schedule endpoint.

**Comment by paddatrapper** [comment]
Hmm... OK. That is problematic because that is how playout determines the next set of media to play

**Comment by jooola** [comment]
Found this https://github.com/jooola/libretime/runs/6947227245?check_suite_focus=true#step:5:373

As the test aren't failing on bionic, maybe the deps didn't get upgraded because of some python constraint.

I'll check if I can fix this.

**Comment by jooola** [comment]
Yup, that was it. I'll open a PR.

**Pull request merged**